### PR TITLE
add upkeep transcoding requirements

### PR DESCRIPTION
**Pull request by se3000**
- also moves all related files into v0.8 directory

### CODE REVIEW

**Comment by se3000** [comment]
After adding a test, I realized that [ethers does not currently play well with enumerables](https://github.com/ethers-io/ethers.js/discussions/2432), so I removed the usage of them.

File: contracts/src/v0.7/dev/UpkeepTranscoderInterface.sol [hunk 1]
```diff
@@ -1,11 +0,0 @@
-// SPDX-License-Identifier: MIT
-
-pragma solidity ^0.8.0;
-
-interface UpkeepTranscoderInterface {
```
Comment (se3000) [inline contracts/src/v0.7/dev/UpkeepTranscoderInterface.sol:5]: This existed in v0.7 and v0.8, but it's pragma was v0.8 in both so it wasn't actually usable in v0.7. I think we should consolidate on one version for now and copy it backwards once the API is finalized.

File: contracts/src/v0.8/dev/KeeperBase.sol [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract KeeperBase {
```
Comment (RyanRHall) [inline contracts/src/v0.8/dev/KeeperBase.sol:4]: we already have a version of this contract `contracts/src/v0.8/KeeperBase.sol`

File: contracts/src/v0.8/dev/KeeperRegistryDev.sol [hunk 3]
```diff
@@ -675,11 +680,23 @@ contract KeeperRegistryDev is
...
+        encodedUpkeeps
+      )
     );
     LINK.transfer(destination, totalBalanceRemaining);
   }
 
+  /**
+   * @inheritdoc MigratableKeeperRegistryInterface
+   */
+  uint8 public constant upkeepTranscoderVersion = UpkeepFormatV1;
```
Comment (RyanRHall) [inline contracts/src/v0.8/dev/KeeperRegistryDev.sol:695]: :+1::+1::+1:

File: contracts/src/v0.8/dev/UpkeepTranscoder.sol [hunk 4]
```diff
@@ -3,31 +3,43 @@
 pragma solidity ^0.8.0;
 
 import "./interfaces/UpkeepTranscoderInterface.sol";
-import "../ConfirmedOwner.sol";
+import "../interfaces/TypeAndVersionInterface.sol";
+
+uint8 constant UpkeepFormatV1 = 1;
```
Comment (RyanRHall) [inline contracts/src/v0.8/dev/UpkeepTranscoder.sol:8]: can we make this an enum?

File: contracts/src/v0.8/dev/interfaces/MigratableKeeperRegistryInterface.sol [hunk 5]
```diff
@@ -1,10 +1,13 @@
 // SPDX-License-Identifier: MIT
 
 pragma solidity ^0.8.0;
-pragma abicoder v2;
+
+import "./UpkeepTranscoderInterface.sol";
```
Comment (RyanRHall) [inline contracts/src/v0.8/dev/interfaces/MigratableKeeperRegistryInterface.sol:5]: Is this import used anywhere?

File: contracts/test/v0.8/UpkeepTranscoder.test.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import { UpkeepTranscoder } from '../../typechain'
+
+let upkeepTranscoderFactory: UpkeepTranscoderFactory
+let transcoder: UpkeepTranscoder
+
+before(async () => {
+  upkeepTranscoderFactory = await ethers.getContractFactory('UpkeepTranscoder')
+})
+
+describe('UpkeepTranscoder', () => {
```
Comment (RyanRHall) [inline contracts/test/v0.8/UpkeepTranscoder.test.ts:14]: thanks for writing the tests for this

**Review by RyanRHall — CHANGES_REQUESTED** [review]

**Pull request merged**